.circleci/config.yml

```diff
@@ -77,42 +78,105 @@ jobs:
       # Note: We need to install versioning library outside of the poetry venv.
       # See note in CONTRIBUTING.md.
-      - run:
+      - run: &build-package
           name: "Build package"
           command: >-
             pip install poetry-dynamic-versioning &&
             poetry build
 
+    environment:
+      GROAPI_TOKEN: dummytoken
+
+  merge-build:
+    <<: *build-job
+    steps:
+      - add_ssh_keys:
+          <<: *ssh-keys
+      - checkout:
+          <<: *checkout-code
+      - run:
+          <<: *setup-environment
+      - run:
+          <<: *run-doctests
+      - store_test_results:
+          <<: *store-test-results
+      - store_artifacts:
+          <<: *store-coverage-report
+      - run:
+          <<: *test-sample-notebook
+      - run:
+          <<: *build-docs
+      - run:
+          <<: *test-docs
+      - run:
+          <<: *publish-docs
+      - run:
+          <<: *build-package
       # Publish new development updates to TestPyPI whenever changes are merged
       # to development.
       # We only want to publish a new package when PRs are actually accepted and
       # merged, which is why we check $CIRCLE_PULL_REQUEST.
       - run:
           name: "Push to testpypi"
           command: >-
-            if [ "$CIRCLE_BRANCH" == "development" ] && [ -z "$CIRCLE_PULL_REQUEST" ]; then
               poetry config repositories.testpypi https://test.pypi.org/legacy/ &&
               poetry publish -u __token__ -p $TESTPYPI_TOKEN -r testpypi
-            fi
 
+
+  release-build:
+    <<: *build-job
+    steps:
+      - add_ssh_keys:
+          <<: *ssh-keys
+      - checkout:
+          <<: *checkout-code
+      - run:
+          <<: *setup-environment
+      - run:
+          <<: *run-doctests
+      - store_test_results:
+          <<: *store-test-results
+      - store_artifacts:
+          <<: *store-coverage-report
+      - run:
+          <<: *test-sample-notebook
+      - run:
+          <<: *build-docs
+      - run:
+          <<: *test-docs
+      - run:
+          <<: *publish-docs
+      - run:
+          <<: *build-package
       # Publish new releases to PyPI.
       # We only check whether this is a tagged build and if so, publish to pypi
       # Check that $CIRCLE_TAG is not a null string
       - run:
           name: "Push to pypi"
           command: >-
-            if [ -n "$CIRCLE_TAG" ]; then
               poetry publish -u __token__ -p $PYPI_TOKEN
-            fi
-
-    environment:
-      GROAPI_TOKEN: dummytoken
 
 workflows:
-  ci-workflows:
+  pull-request:
     jobs:
-      - ci:
+      - build:
           filters:
-            tags:
-              only: /^v.*/
+            branches:
+              ignore: "/^development$/"
+
 
+  merge-commit:
+    jobs:
+      - merge-build:
+          filters:
+            branches:
+              only: "/^development$/"
+
+  release:
+    jobs:
+      - release-build:
+          filters:
+            branches:
+              ignore: "/.*/"
```

Comment:
Does this signify ignore changes to every branch?
Combined with the tags line does it mean only run when a tag is added?

Reply:
Yes, combined it means only run with every new tag.